packages/flutter/lib/src/rendering/flex.dart

```diff
@@ -7,6 +7,15 @@ import 'dart:math' as math;
 import 'box.dart';
 import 'object.dart';
 
+/// Whether to apply tight or loose constraints to flexible children.
+enum FlexConstraintType {
+  /// Require the child to fill the flexible space.
+  tight,
+
+  /// Permit the child to use less than all the flexible space.
```

Comment:
mention that the space becomes "free" space and isn't allocated to the other children if all the space isn't used